extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -691,9 +706,27 @@ export class AmpStoryPage extends AMP.BaseElement {
       if (this.isBotUserAgent_) {
         mediaEl.muted = false;
         mediaEl.removeAttribute('muted');
+        if (mediaEl.tagName === 'AUDIO' && mediaEl.paused) {
+          mediaEl.play();
+        }
       } else {
-        return mediaPool.unmute(
-            /** @type {!./media-pool.DomElementDef} */ (mediaEl));
+        mediaEl = /** @type {!./media-pool.DomElementDef} */ (mediaEl);
+        const promises = [mediaPool.unmute(mediaEl)];
+
+        // Audio element might not be playing if the page navigation did not
+        // happen after a user intent, and the media element was not "blessed".
+        // On unmute, make sure this audio element is playing, at the expected
+        // currentTime.
+        if (mediaEl[PLAY_AUDIO_ELEMENT_FROM_TIMESTAMP_PROPERTY_NAME]) {
+          const timeStart =
+              mediaEl[PLAY_AUDIO_ELEMENT_FROM_TIMESTAMP_PROPERTY_NAME];
+          let currentTime = (Date.now() - timeStart) / 1000;
+          mediaEl.duration && (currentTime = currentTime % mediaEl.duration);
```

Comment:
This is crafty but we should probably just write it the normal way 😄 

```javascript
if (mediaEl.duration) {
  mediaEl.currentTime = currentTime % mediaEl.duration;
}
```

The actual logic LGTM

Reply:
Done